cg/utils/files.py

```diff
@@ -27,3 +31,14 @@ def get_files_matching_pattern(directory: Path, pattern: str) -> List[Path]:
 def get_all_files_in_dir(base_path: Path) -> Set[Path]:
     """Get a set of all files relative to the given base path."""
     return {file.relative_to(base_path) for file in base_path.rglob("*") if file.is_file()}
+
+
+def rename_file(file_path: Path, renamed_file_path: Path) -> None:
+    """Rename the given fastq file path."""
+    if renamed_file_path.exists():
+        LOG.debug(
+            f"File {renamed_file_path} already exists. Skipping renaming of {renamed_file_path}."
+        )
+        return
+    shutil.move(src=file_path, dst=renamed_file_path)
```

Comment:
I think os.rename is more appropriate. I think shutil is more for large collections of files

Reply:
Another option is to use Pathlib
```python
from pathlib import Path
src = Path("src.txt")
dst = Path("dst.txt")
src.rename(dst)
```

Reply:
Since this function is for generic use I will keep `shutil` since as it is more flexible and allows us to use the function in more places. From what I found:

1 OS module might fail to move a file if the source and destination path is on different file systems or drive.
But shutil.move will not fail in this kind of cases.
2 shutil.move checks if the source and destination path are on the same file system or not. But os.rename does not check, thus it fails sometimes.
3 After checking the source and destination path, if it is found that they are not in the same file system, shutil.move will copy the file first to the destination. Then it will delete the file from the source file. Thus we can say shutil.move is a smarter method to move a file in Python when the source and destination path are not on the same drive or file system.
4 shutil.move works on high-level functions, while os.rename works on lower-level functions.

Do you agree?

Reply:
I think the cases you are referring are edge cases, for which we probably want it to fail (at least with the current usage). I vote using Path as it is a simple solution and a library most are familiar with.

Reply:
Changed to using path.rename()